vault/auth/kubernetes/login.go

```diff
@@ -34,64 +32,58 @@ func init() {
 }
 
 func (ai *AuthInfo) GetLoginToken() (string, error) {
-	url := fmt.Sprintf("%s/%s/login", ai.vaultUrl, path)
-
-	jwt, err := GetJWT(ai.pod.ServiceAccount, ai.pod.Namespace)
+	kubeClient, err := getKubeClient()
 	if err != nil {
 		return "", err
 	}
-
-	body := map[string]interface{}{
-		"role": ai.authRole,
-		"jwt":  jwt,
-	}
-	data, err := json.Marshal(body)
+	app, err := getAppBinding(ai.refName, ai.refNamespace)
 	if err != nil {
 		return "", err
 	}
-
-	secret, err := getLoginSecret(url, data)
+	binding := app.DeepCopy()
+	binding.Spec.Secret.Name, err = getServiceAccountSecret(kubeClient, ai.pod.ServiceAccount, ai.pod.Namespace)
```

Comment:
Update `binding` namespace, just in case pod's service account is in different namespace than the appbinding